src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1180,8 +1180,7 @@ public int Send(IList<ArraySegment<byte>> buffers, SocketFlags socketFlags, out
                 // Don't log transfered byte count in case of a failure.
                 return 0;
             }
-
-            if (SocketsTelemetry.Log.IsEnabled())
+            else if (SocketsTelemetry.Log.IsEnabled())
```

Comment:
Forgive me my stupid question, but how does the `else if` instead of just `if` here helps? If the previous block ends with `return`.

Reply:
I did it uniformly in all cases for the sake of a higher readability. There are other places without `return` inside the `if (errorCode != SocketError.Success)`, but we still don't want to log anything there in case of a failure. ([one example](https://github.com/dotnet/runtime/blob/891dc7fc2e53acd65c40a7545567bba414a992b7/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs#L1280))